src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,267 @@
+            flightPlan[it.flight!!] = payload
+            tableColsToPrint[it.flight!!] = it.sourcePrimaryKeyColumns
+        }
+        val destinationsMap = generateDestinationsMap(integration, missionParameters, dataIntegrationApi)
+
+        val shuttle = Shuttle(
+                integration.environment,
+                flightPlan,
+                entitySets.values.associateBy { it.name },
+                entityTypes.values.associateBy { it.id },
+                propertyTypes.values.associateBy { it.type },
+                destinationsMap,
+                dataIntegrationApi,
+                tableColsToPrint,
+                missionParameters,
+                StorageDestination.S3,
+                blackbox,
+                Optional.of(entitySets.getValue(integration.logEntitySetId.get())),
+                Optional.of(jobId),
+                Optional.of(integrationJob.second),
+                idService,
+                hazelcastInstance
+        )
+
+        executor.submit {
+            semaphore.acquire()
+            shuttle.launch(uploadBatchSize)
+        }.addListener(Runnable { semaphore.release() }, executor)
+    }
+
+    fun pollIntegrationStatus(jobId: UUID): IntegrationStatus {
+        checkState(integrationJobs.containsKey(jobId), "Job Id $jobId is not assigned to an existing integration job")
+        val status = integrationJobs.getValue(jobId).integrationStatus
+        if (status == IntegrationStatus.SUCCEEDED || status == IntegrationStatus.FAILED) {
+            integrationJobs.remove(jobId)
+        }
+        return status
+    }
+
+    fun pollAllIntegrationStatuses(): Map<UUID, IntegrationJob> {
+        return integrationJobs
+    }
+
+    fun createIntegrationDefinition(integrationName: String, integration: Integration) {
+        checkState(!integrations.containsKey(integrationName), "An integration with name $integrationName already exists.")
+        val logEntitySetId = createLogEntitySet(integrationName, integration)
+        integration.logEntitySetId = Optional.of(logEntitySetId)
+        integrations[integrationName] = integration
+    }
+
+    fun readIntegrationDefinition(integrationName: String): Integration {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
+        return integrations.getValue(integrationName)
+    }
+
+    fun updateIntegrationDefinition(integrationName: String, integrationUpdate: IntegrationUpdate) {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
+        integrations.executeOnKey(integrationName, UpdateIntegrationEntryProcessor(integrationUpdate))
+    }
+
+    fun deleteIntegrationDefinition(integrationName: String) {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
```

Comment:
since this check is repeated, it might be worth putting it into a helper like
```
private fun checkIntegrationExists(integrationName: String) {
  checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
}
```

and then you only need to call the helper